dali/kernels/imgproc/convolution/convolution_cpu_test.cc

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <chrono>
+#include <cmath>
+#include <complex>
+#include <tuple>
+#include <vector>
+
+#include "dali/core/boundary.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/convolution_cpu.h"
+#include "dali/kernels/scratch.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+
+template <typename T>
+struct CyclicWindowWrapperTest : public ::testing::Test {};
+
+TYPED_TEST_SUITE_P(CyclicWindowWrapperTest);
+
+template <int num_lanes_>
+struct cww_params {
+  static constexpr int num_lanes = num_lanes_;
+};
+
+using CyclicWindowWrapperValues = ::testing::Types<cww_params<1>, cww_params<3>, cww_params<16>>;
+
+TYPED_TEST_P(CyclicWindowWrapperTest, FillAndCycle) {
+  constexpr int size = 6;
+  constexpr int num_lanes = TypeParam::num_lanes;
+  int tmp_buffer[size * num_lanes];    // NOLINT
+  int input_buffer[size * num_lanes];  // NOLINT
+  for (int i = 0; i < size * num_lanes; i++) {
+    input_buffer[i] = i;
+    tmp_buffer[i] = -1;
+  }
+  CyclicWindowWrapper<int, 16> cww(tmp_buffer, size, num_lanes);
+  EXPECT_EQ(0, cww.Size());
+  for (int i = 0; i < size; i++) {
+    cww.PushElement(input_buffer + i * num_lanes);
+    EXPECT_EQ(tmp_buffer + i * num_lanes, cww.GetElementOffset(i));
+    for (int c = 0; c < num_lanes; c++) {
+      EXPECT_EQ(input_buffer[i * num_lanes + c], cww.GetElementOffset(i)[c]);
+    }
+  }
+  for (int i = 0; i < size; i++) {
+    cww.PopElement();
+    cww.PushElement(input_buffer + i * num_lanes);
+    for (int j = 0; j < size; j++) {
+      // we're starting at i + 1 as we did already one Pop & Push operation
+      int element = (i + 1 + j) % size;
+      EXPECT_EQ(tmp_buffer + element * num_lanes, cww.GetElementOffset(j));
+      for (int c = 0; c < num_lanes; c++) {
+        EXPECT_EQ(input_buffer[element * num_lanes + c], cww.GetElementOffset(j)[c]);
+      }
+    }
+  }
+}
+
+void baseline_dot(span<int> result, span<const int> input, span<const int> window, int in_offset) {
```

Comment:
```suggestion
void BaselineDot(span<int> result, span<const int> input, span<const int> window, int in_offset) {
```
nitpick: consistency with the rest of the file